src/skibidi_orm/migration_engine/schema_analysys/state_manager.py

```diff
@@ -0,0 +1,125 @@
+from skibidi_orm.migration_engine.adapters.base_adapter import (
+    BaseTable,
+    BaseColumn,
+    Relation,
+)
+from typing import Any
+from skibidi_orm.migration_engine.operations.column_operations import (
+    ColumnOperation,
+    AddColumnOperation,
+    DeleteColumnOperation,
+)
+
+from skibidi_orm.migration_engine.operations.table_operations import (
+    TableOperation,
+    CreateTableOperation,
+    DeleteTableOperation,
+    RenameTableOperation,
+)
+
+
+class StateManager[TTable: BaseTable[BaseColumn[Any]]]:
+    def __init__(
+        self,
+        db_tables: list[TTable],
+        db_relations: list[Relation],
+        schema_tables: list[TTable],
+        schema_relations: list[Relation],
+    ) -> None:
+        self.db_tables = db_tables
+        self.db_relations = db_relations
+        self.schema_tables = schema_tables
+        self.schema_relations = schema_relations
+
+        self.serviced_tables: list[TTable] = []
+
+        self.operations: list[TableOperation | ColumnOperation] = []
+
+        self.analyze_schemas()
+
+    def analyze_schemas(self):
+        self.get_delete_tables_operations()
+        self.get_create_tables_operations()
+        self.get_delete_columns_operations()
+        self.get_create_columns_operations()
+
+    def get_operations(self) -> list[TableOperation | ColumnOperation]:
+        return self.operations
+
+    def get_create_tables_operations(self) -> None:
+
+        t_db_names = set([t.name for t in self.db_tables])
+
+        for s_table in self.schema_tables:
+            if s_table.name not in t_db_names:
+                if s_table not in self.serviced_tables:
+                    self.operations.append(CreateTableOperation(s_table))
+                    self.serviced_tables.append(s_table)
+
+    def get_delete_tables_operations(self) -> None:
+
+        class TableFound(Exception):
```

Comment:
i think defining an exception type inside of a method is a bit messy, maybe move it out?

Reply:
why do we use it to control the flow of the program anyway? can't we create a helper function containing the inner loop and simply return from it if the table is found, else after adding the deleteoperation?